From 6c833c1dd912e8cc3a8befd9ed0fa9f0c9f6a50d Mon Sep 17 00:00:00 2001
From: Andrew Sutherland <dr3wsuth3rland@gmail.com>
Date: Wed, 14 Oct 2015 17:55:56 -0500
Subject: [PATCH] add netgo tag

---
 build.go | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/build.go b/build.go
index 2b28985..30a1c50 100644
--- a/build.go
+++ b/build.go
@@ -35,6 +35,7 @@ var (
 	goarch    string
 	goos      string
 	noupgrade bool
+	netgo     bool
 	version   string
 	goVersion float64
 	race      bool
@@ -60,6 +61,7 @@ func main() {
 	flag.StringVar(&goarch, "goarch", runtime.GOARCH, "GOARCH")
 	flag.StringVar(&goos, "goos", runtime.GOOS, "GOOS")
 	flag.BoolVar(&noupgrade, "no-upgrade", noupgrade, "Disable upgrade functionality")
+	flag.BoolVar(&netgo, "netgo", netgo, "Enable netgo build tag")
 	flag.StringVar(&version, "version", getVersion(), "Set compiled in version string")
 	flag.BoolVar(&race, "race", race, "Use race detector")
 	flag.Parse()
@@ -78,6 +80,9 @@ func main() {
 		if noupgrade {
 			tags = []string{"noupgrade"}
 		}
+		if netgo {
+			tags = append(tags, "netgo")
+		}
 		install("./cmd/...", tags)
 
 		vet("./cmd/syncthing")
@@ -98,6 +103,9 @@ func main() {
 			if noupgrade {
 				tags = []string{"noupgrade"}
 			}
+			if netgo {
+				tags = append(tags, "netgo")
+			}
 			install(pkg, tags)
 
 		case "build":
@@ -106,6 +114,9 @@ func main() {
 			if noupgrade {
 				tags = []string{"noupgrade"}
 			}
+			if netgo {
+				tags = append(tags, "netgo")
+			}
 			build(pkg, tags)
 
 		case "test":
@@ -242,6 +253,10 @@ func buildTar() {
 		tags = []string{"noupgrade"}
 		name += "-noupgrade"
 	}
+	if netgo {
+		tags = append(tags, "netgo")
+		name += "-netgo"
+	}
 	build("./cmd/syncthing", tags)
 	filename := name + ".tar.gz"
 	files := []archiveFile{
@@ -270,6 +285,10 @@ func buildZip() {
 		tags = []string{"noupgrade"}
 		name += "-noupgrade"
 	}
+	if netgo {
+		tags = append(tags, "netgo")
+		name += "-netgo"
+	}
 	build("./cmd/syncthing", tags)
 	filename := name + ".zip"
 	files := []archiveFile{
-- 
2.5.2

