From b73d8feff0d37b2684c5bc5a4019784a49ec9c4e Mon Sep 17 00:00:00 2001
From: Andrew Sutherland <dr3wsuth3rland@gmail.com>
Date: Tue, 7 Apr 2015 09:28:12 -0500
Subject: [PATCH 1/2] gimme some pie

---
 build.go | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/build.go b/build.go
index 54d22b6..706df47 100644
--- a/build.go
+++ b/build.go
@@ -449,6 +449,9 @@ func ldflags() string {
 	fmt.Fprintf(b, " -X main.BuildUser%c%s", sep, buildUser())
 	fmt.Fprintf(b, " -X main.BuildHost%c%s", sep, buildHost())
 	fmt.Fprintf(b, " -X main.BuildEnv%c%s", sep, buildEnvironment())
+	if goos == "android" {
+		b.WriteString(` -extldflags '-fPIE -pie'`)
+	}
 	return b.String()
 }
 
-- 
2.5.2

