From d4fb633e21f0b4ef5e82b83dc9535f52afd5a562 Mon Sep 17 00:00:00 2001
From: Andrew Sutherland <dr3wsuth3rland@gmail.com>
Date: Tue, 7 Apr 2015 09:28:53 -0500
Subject: [PATCH 2/2] usage: remove android workaround

---
 cmd/syncthing/usage_report.go | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/cmd/syncthing/usage_report.go b/cmd/syncthing/usage_report.go
index 7b1f830..abe3f45 100644
--- a/cmd/syncthing/usage_report.go
+++ b/cmd/syncthing/usage_report.go
@@ -13,14 +13,12 @@ import (
 	"crypto/tls"
 	"encoding/json"
 	"fmt"
-	"net"
 	"net/http"
 	"runtime"
 	"sort"
 	"time"
 
 	"github.com/syncthing/syncthing/lib/config"
-	"github.com/syncthing/syncthing/lib/dialer"
 	"github.com/syncthing/syncthing/lib/model"
 	"github.com/syncthing/syncthing/lib/protocol"
 	"github.com/syncthing/syncthing/lib/upgrade"
@@ -251,12 +249,6 @@ func (s *usageReportingService) sendUsageReport() error {
 
 	transp := &http.Transport{}
 	client := &http.Client{Transport: transp}
-	if BuildEnv == "android" {
-		// This works around the lack of DNS resolution on Android... :(
-		transp.Dial = func(network, addr string) (net.Conn, error) {
-			return dialer.Dial(network, "194.126.249.13:443")
-		}
-	}
 
 	if s.cfg.Options().URPostInsecurely {
 		transp.TLSClientConfig = &tls.Config{
-- 
2.5.2

